packages/react/src/primitives/utils/useStableId.ts

```diff
@@ -0,0 +1,31 @@
+// Adapted from https://github.com/radix-ui/primitives/blob/main/packages/react/id/src/id.tsx#L8
+import * as React from 'react';
+import { useLayoutEffect } from './useLayoutEffect';
+
+// Prefixed autogenerated id created by useStableId
+export const AUTO_GENERATED_ID_PREFIX = 'amplify-id';
+
+export type UseStableId = (id?: string) => string;
+
+// Create a local version of React.useId which will reference React.useId for React 18
+// and fallback to noop for React 17 and below
+const useReactId = (React as any).useId || (() => undefined);
```

Comment:
Also, what about this? We can keep type safety if `useId` is present.
```suggestion
const useReactId = 'useId' in React ? React.useId : (() => undefined);
```

Reply:
Looks great, will update.

Reply:
So, since we're still using React 17, and it doesn't have useId yet, and this causes a type error. I think I'd prefer to typecast as opposed to disabling type checking on this line. I think the risk is pretty low in this case. What do you think?

Reply:
I'm not a fan of ignoring types, but if there's no way to do it, i'm good